cpu/src/lib.rs

```diff
@@ -60,13 +65,26 @@ where
         RowMajorMatrix::new(rows.concat(), NUM_CPU_COLS)
     }
 
-    fn generate_permutation_trace(
-        &self,
-        machine: &M,
-        main_trace: RowMajorMatrix<Self::F>,
-        random_elements: Vec<Self::FE>,
-    ) -> RowMajorMatrix<Self::F> {
-        todo!()
+    fn global_sends(&self, machine: &M) -> Vec<Interaction<M::F>> {
+        let mem_sends = (0..3).map(|i| {
+            let channel = &CPU_COL_INDICES.mem_channels[i];
+            let is_read = VirtualPairCol::single_main(channel.is_read);
+            let addr = VirtualPairCol::single_main(channel.addr);
+            let value = channel.value.0.map(|v| VirtualPairCol::single_main(v));
+            let mut fields = vec![is_read, addr];
+            fields.extend(value);
+            Interaction {
+                fields,
+                count: VirtualPairCol::single_main(channel.used),
+                argument_index: machine.mem_bus(),
+            }
+        });
+        let send_general = Interaction {
+            fields: vec![],
+            count: VirtualPairCol::single_main(CPU_COL_INDICES.opcode_flags.is_bus_op),
+            argument_index: 0,
```

Comment:
This should be `machine.general_bus()`, correct? And in `fields`, we would be sending `ChipChannelCols`?

Reply:
Yes fixed, thanks